diff --git a/src/core/dns.c b/src/core/dns.c
index 3ba88db0..38be6510 100644
--- a/src/core/dns.c
+++ b/src/core/dns.c
@@ -84,7 +84,7 @@
 
 /** DNS server IP address */
 #ifndef DNS_SERVER_ADDRESS
-#define DNS_SERVER_ADDRESS        inet_addr("208.67.222.222") /* resolver1.opendns.com */
+#define DNS_SERVER_ADDRESS        inet_addr("8.8.8.8") /* Google DNS */
 #endif
 
 /** DNS server port address */