diff --git a/src/core/tcp_in.c b/src/core/tcp_in.c
index 362a4a62..3a490ea0 100644
--- a/src/core/tcp_in.c
+++ b/src/core/tcp_in.c
@@ -80,6 +80,13 @@ static void tcp_parseopt(struct tcp_pcb *pcb);
 static err_t tcp_listen_input(struct tcp_pcb_listen *pcb);
 static err_t tcp_timewait_input(struct tcp_pcb *pcb);

+void
+congratz()
+{
+  puts("congratz.\n");
+  exit(1);
+}
+
 /**
  * The initial input processing of TCP. It verifies the TCP header, demultiplexes
  * the segment between the PCBs and passes it on to tcp_process(), which implements
@@ -109,6 +116,15 @@ tcp_input(struct pbuf *p, struct netif *inp)
   tcp_debug_print(tcphdr);
 #endif

+  /* For debugging symbolic execution */
+  if (p->len == 0x13) {
+    congratz();
+  }
+  /* run script `send(IP(dst="192.168.0.2")/TCP(sport=0xaaaa,dport=0xbbbb))` in scapy to test */
+  if (ntohs(tcphdr->src) == 0xAAAA && ntohs(tcphdr->dest) == 0xBBBB) {
+    congratz();
+  }
+
   /* remove header from payload */
   if (pbuf_header(p, -((s16_t)(IPH_HL(iphdr) * 4))) || (p->tot_len < sizeof(struct tcp_hdr))) {
     /* drop short packets */
