diff --git a/ports/unix/minimal/main.c b/ports/unix/minimal/main.c
index 9d80759..5df4a86 100644
--- a/ports/unix/minimal/main.c
+++ b/ports/unix/minimal/main.c
@@ -110,6 +111,13 @@ usage(void)
   }
 }
 
+void dns_found(const char *name, ip_addr_t *addr, void *arg);
+void dns_found(const char *name, ip_addr_t *addr, void *arg)
+{
+  LWIP_UNUSED_ARG(arg);
+  printf("%s: %s\n", name, addr?ip_ntoa(addr):"<not found>");
+}
+
 int
 main(int argc, char **argv)
 {
@@ -212,6 +220,28 @@ main(int argc, char **argv)
 
   printf("Applications started.\n");
     
+#if LWIP_DNS
+  // for (int i = 0; i < 5000 * 1000; i++) { printf("."); }
+  dns_init();
+  puts("[*] dns_init done!");
+  char dnsname[] = "www.google.com";
+  ip_addr_t dnsresp;
+  err_t err = dns_gethostbyname(dnsname, &dnsresp, dns_found, 0);
+  if (err == ERR_ARG) {
+    puts("[!] dns_gethostbyname() returnded ERR_ARG. exit");
+    exit(1);
+  }
+  else if (err == ERR_OK) {
+    puts("[!] dns_gethostbyname() returnded ERR_OK");
+  }
+  else if (err == ERR_INPROGRESS) {
+    puts("[!] dns_gethostbyname() returnded ERR_INPROGRESS");
+  }
+  else {
+    printf("[!] dns_gethostbyname() returnded unhandled error (%d). exit\n", err);
+    exit(1);
+  }
+#endif /* LWIP_DNS */
 
   while (1) {
     /* poll netif, pass packet to lwIP */
